### PR TITLE
refactor(directiveLifecycle): switch test from TCB to use TestBed

### PR DESCRIPTION
**Pull request by cnishina**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format

**What kind of change does this PR introduce?**
Refactoring (no functional changes, no api changes)

**What is the current behavior?**
Using deprecated TestComponentBuilder and component directives

**What is the new behavior?**
Use TestBed

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

**Comment by IgorMinar** [comment]
@cnishina please rebase

**Comment by cnishina** [comment]
@IgorMinar done.

File: modules/@angular/core/test/directive_lifecycle_integration_spec.ts [hunk 1]
```diff
@@ -9,40 +9,45 @@
...
+import {TestBed} from '@angular/core/testing';
 import {AsyncTestCompleter, Log, TestComponentBuilder, beforeEach, beforeEachProviders, ddescribe, describe, expect, iit, inject, it, xdescribe, xit} from '@angular/core/testing/testing_internal';
 
 export function main() {
   describe('directive lifecycle integration spec', () => {
-
+    let log: Log;
     beforeEachProviders(() => { return [Log]; });
 
+    beforeEach(() => {
```
Comment (juliemr) [inline modules/@angular/core/test/directive_lifecycle_integration_spec.ts]: `compileComponents()` should always be in an `async`

Comment (juliemr) [inline modules/@angular/core/test/directive_lifecycle_integration_spec.ts]: Or, you can remove it alltogether if there's no template. I'd recommend that here.

**Comment by juliemr** [comment]
Please address one comment, then this is ready to go.

**Comment by juliemr** [comment]
Lint is upset, missing semicolon

**Comment by IgorMinar** [comment]
please squash

**Comment by cnishina** [comment]
Done.

**Pull request merged**